eth/stagedsync/stage_execute.go

```diff
@@ -73,8 +77,11 @@ func SpawnExecuteBlocksStage(s *StageState, stateDB ethdb.Database, chainConfig
 		defer tx.Rollback()
 	}
 
-	batch := tx.NewBatch()
-	defer batch.Rollback()
+	var batch ethdb.DbWithPendingMutations
+	if params.SilkwormExecutionFunc == nil {
```

Comment:
I wonder if we can instead have the whole different execution stage instead of big if/else in the current one and additional parameter?

Reply:
you can walk through `stagedsync.DefaultStages()` and find the `StageBuilder` with id ==`stages.Execution` and just replace it with something else. I was even planning to add a helper method for that but didn't yet get to it.

Reply:
Well, I'd like to keep Silkworm and non-Silkworm execution next to each other so that discrepancies are minimised. Let me refactor SpawnExecuteBlocksStage a little bit to make it cleaner and more unified between Silkworm and go execution.

Reply:
Makes sense! Let’s do that.